draft-jmiller-json-proof-algorithms.md

```diff
@@ -363,6 +365,59 @@ The same JWP in compact serialization:
 ImV5SnBjM01pT2lKb2RIUndjem92TDJsemMzVmxjaTVsZUdGdGNHeGxJaXdpWTJ4aGFXMXpJanBiSW1aaGJXbHNlVjl1WVcxbElpd2laMmwyWlc1ZmJtRnRaU0lzSW1WdFlXbHNJaXdpWVdkbElsMHNJblI1Y0NJNklrcFFWQ0lzSW1Gc1p5STZJa0pDVXkxWUluMCI.~IkpheSI~~NDI.AAUVqUnyMW_qGrrCzK7xHqUydeyQX5O6276HzUKz41aX9q7qzC6r-eHGh1GD4Y3ZL9l_sA9ibulK3OYbyXVy0ym3LEC5LYyZMUnHXdD41K9qx1P_gE2Y-qZlTvgKauwY7hUihoC6hJl7Vlic9ssX_fjZ6QGo0AzB3o5aHN_xgqSQUwAPpRmckfMnWPn2uZgD3Ei0AAAAdIW0OjVpeHzjoE7lStFvpGW3VnrUfloX5G24SUtysY6yWWtkvbtKj6fausG99w6GKAAAAAIFgnhW_xwhkRSVwwgEyNSyQ5PmrsAJnl6zkD88Uv_YBFVs0cLRsWpF1-uxU_QBw2aHYxR5B_wau84Q-oYB_8VcgmnxryMW0mWecdbeAwSovPsi1dPglpMmpOWX4t-8tbTM5DprN-iUtMcqtX876eq8AAAABF3EHyaXaecu5C5WxIjUr6pTFU4T4NN6B1xHEatCejiCLROs2UE_9icGHk2EViQpA-pIkshllp9sjA_DOfmav8xbHp8gnRgDblpmY84qOgG111UdIIMuTBkFK8sg16epbDiukjOuKL4WJV2c9dAaN7SHc0ZgavPVg8Q_pSqdFl4u
 ```
 
+
+## Salted HMAC
+
+The Salted HMAC (SH) algorithm is similar to Single Use (SU), but uses generated salts and HMAC instead of individual JWSs to protect each payload.  It requires less compute but can result in potentially larger presentation proof values.
+
+Like SU, it also does not support unlinkability if the same JWP is presented multiple times and requires an individually issued JWP for each presentation in order to fully protect privacy.
+
+### SU-based Setup
+
+This algorithm uses most of the same setup as the Single Use one, including the use of JWS as the signature mechanism and the holder providing a Presentation Key.
+
+The main difference with the SH algorithm is that instead of the issuer generating an Ephemeral Key for signing, it instead generates a Shared Secret using HMAC.  This Shared Secret is then used to generate unique salt values that become the input keys to HMAC the issuer protected header and each payload.
+
+### Issuer Protected Header
+
+The holder's Presentation Key JWK MUST be included in the issuer protected header using the `presentation_jwk` claim.  The issuer MUST validate that the holder has posession of this key through a trusted mechanism such as verifying the signature of a unique nonce value from the holder.
```

Comment:
We discussed whether to seperate out two algorithm registrations here one that features holder binding the other does not. Im assuming by this text you would rather collapse this into one and just require implementers to validate the presence of this claim to detect whether it is a holder bound version?

Reply:
I'm just starting with the stronger holder-bound version.

I really don't want to create a JPA that lacks support for binding or replay protection if we don't have to.  The argument is the same as alg:"none", that it's useful when the channel or context provides the protections, but if we can avoid that here then we minimize the chance for people to implement or use it incorrectly.